runners/flink/1.15/src/main/java/org/apache/beam/runners/flink/transform/sql/TableInfo.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.transform.sql;
+
+import java.io.Serializable;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.types.AbstractDataType;
+
+/**
+ * A package private container class holding the classes with Table information.
+ *
+ * <ul>
+ *   <li>{@link InputTableInfo InputTableInfo} contains information for the main input table of a
+ *       Sql Transform..
+ *   <li>{@link AdditionalInputTableInfo InputTableInfo} contains information for an additional
+ *       input table of a Sql Transform.
+ *   <li>{@link OutputTableInfo InputTableInfo} contains information for the main output table of a
+ *       Sql transform.
+ *   <li>{@link AdditionalOutputTableInfo InputTableInfo} contains information for an additional
+ *       output table of a Sql Transform.
+ * </ul>
+ */
+class TableInfo {
+  public static class InputTableInfo<T> implements Serializable {
```

Comment:
Seems we can consolidate both InputTableInfo and AdditionalInputTableInfo into one if we have the TupleTag of InputTableInfo. We should be able to construct it based on the inputClass and default main input name.